### PR TITLE
rotting meat

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
- most raw meat can now rot:
  - 5 mins it will say rotting in examine
  - 10 mins it will turn into rotten meat
- rotten meat is now actually bad (less proteins, has toxin so its evil)
- putting raw meat in a freezer or cold place/space will prevent/slow rotting
- rotting meat wont produce miasma since thats bad
- cutlets are immune because cbf to make rotten cutlet sprite and ig its cured or something
- xeno meat doesnt rot because le acid lore
- ian meat doesnt rot because that would be incredibly shitty
- raw bacon doesnt rot because its cured

- added RotIntoComponent and FoodMeatRawBase
- rotting dead check only applies if the perishable thing has MobState in the first place to let non-mobs rot

rotting times and the component name ***subject to bikeshedding*** but overall this will make chef gameplay more engaging and realistic

**Media**
1 min in: (will be 5 now)

2 mins in: (will be 10 now)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- tweak: Most raw meat will now rot outside of a freezer. Please don't eat rotten meat.

### CODE REVIEW

**Comment by UbaserB** [comment]
me like

**Comment by DogZeroX** [comment]
Me when I take meat out of the freezer, put it on the table and it's rotting already. The timer is way too small and it'll be incredibly annoying.

**Comment by metalgearsloth** [comment]
Making it 10 like bodies is probably a good start.

**Comment by deltanedas** [comment]
with how rotting works it would mean 20 minutes until it fully rots is that still good
or should i have it be 7.5mins then 15 minutes to fully rot, plenty of time to play with it but cant just leave it lying around

**Comment by deltanedas** [comment]
@metalgearsloth are numbers ok now or does it need to be slower

**Comment by deltanedas** [comment]
@metalgearsloth j

File: Content.Shared/Atmos/Miasma/RotIntoComponent.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public sealed class RotIntoComponent : Component
+{
+    /// <summary>
+    /// Entity to rot into.
+    /// </summary>
+    [DataField("entity", required: true, customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>)), ViewVariables(VVAccess.ReadWrite)]
+    public string Entity = string.Empty;
+
+    /// <summary>
+    /// Rotting stage to turn at.
```
Comment (metalgearsloth) [inline Content.Shared/Atmos/Miasma/RotIntoComponent.cs:21]:
```suggestion
    /// Rotting stage to turn at, this is a multiplier of the total rot time.
```

**Comment by metalgearsloth** [comment]
pro

**Comment by deltanedas** [comment]
real

**Comment by Fey-the-Bin** [comment]
So if I put raw meat inside a fridge, and the fridge is inside the walk-in freezer, would the meat be prevented/slowed from rotting?

**Comment by deltanedas** [comment]
fridge itself prevents rotting like morgues and cold areas/space

**Comment by metalgearsloth** [comment]
the rot consumes jimbo

**Pull request merged**